src/promptflow-core/promptflow/executor/_service/middlewares/custom_float_middleware.py

```diff
@@ -0,0 +1,30 @@
+import json
+
+from fastapi.responses import StreamingResponse
+from starlette.middleware.base import BaseHTTPMiddleware
+from starlette.requests import Request
+
+from promptflow._constants import DEFAULT_ENCODING
+
+
+class CustomFloatMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        response = await call_next(request)
+
+        # Ensure the response content type is JSON
+        if not isinstance(response, StreamingResponse) and "application/json" in response.headers.get(
+            "content-type", ""
+        ):
+            data = json.loads(response.body)
+            # Modify the JSON data by handling NaN values and serialize it back to JSON string
+            modified_body = json.dumps(data, default=self.handle_nan)
+            # Update the response body with the new JSON data
+            response.body = modified_body.encode(DEFAULT_ENCODING)
+
+        return response
+
+    def handle_nan(self, obj):
+        # Converts float NaN values to None (which becomes "NaN" in JSON)
+        if isinstance(obj, float) and obj != obj:
+            return str(obj)
```

Comment:
So nan would be "NaN"?

Does it mean that the output type would also be changed?